### PR TITLE
fix(terraform): fix root module search

### PR DESCRIPTION
**Pull request by nikpivkin**
## Description

Modules with count = 0 were counted as root modules. This PR changes the way root modules are defined.

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/5665

## Related PRs:
- https://github.com/aquasecurity/trivy/pull/6161

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/iac/scanners/terraform/parser/modules_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			name: "multiple root modules",
+			files: map[string]string{
+				"code/main.tf": `
+module "this" {
+  count = 0
+  source = "./modules/s3"
+}`,
+				"code/modules/s3/main.tf": `
+module "this" {
+  source = "./modules/logging"
```
Comment (simar7) [inline pkg/iac/scanners/terraform/parser/modules_test.go:31]: Does the parsing logic still work if this was:

```suggestion
  source = "modules/logging"
```

It is possible to define file paths without mentioning the present dir.

File: pkg/iac/scanners/terraform/parser/parser_test.go [hunk 2]
```diff
@@ -1139,3 +1139,84 @@ func TestForEachWithObjectsOfDifferentTypes(t *testing.T) {
...
+			require.NoError(t, err)
+			assert.Len(t, modules, 1)
+
+			resources := modules.GetResourcesByType("test")
+			assert.Len(t, resources, tt.expected)
+		})
+	}
+}
+
+func TestCountMetaArgumentInModule(t *testing.T) {
```
Comment (simar7) [inline pkg/iac/scanners/terraform/parser/parser_test.go:1183]: Can we refactor this and make into a table driven test?

File: pkg/iac/scanners/terraform/scanner_test.go [hunk 3]
```diff
@@ -1321,3 +1321,72 @@ deny[res] {
...
+	require.NoError(t, err)
+
+	require.Len(t, results, 1)
+
+	failed := results.GetFailed()
+
+	assert.Len(t, failed, 1)
+
+	occurrences := failed[0].Occurrences()
+	assert.Equal(t, "code/example/main.tf", occurrences[len(occurrences)-1].Filename)
```
Comment (simar7) [inline pkg/iac/scanners/terraform/scanner_test.go:1391]: simplify

```suggestion
	assert.Equal(t, "code/example/main.tf", occurrences[0].Filename)
```

File: pkg/iac/scanners/terraform/parser/modules_test.go (hunk 1 above)
Comment (nikpivkin) [reply pkg/iac/scanners/terraform/parser/modules_test.go:31]: >A local path must begin with either ./ or ../ to indicate that a local path is intended, to distinguish from [a module registry address](https://developer.hashicorp.com/terraform/language/modules/sources#terraform-registry).

https://developer.hashicorp.com/terraform/language/modules/sources#local-paths

File: pkg/iac/scanners/terraform/scanner_test.go (hunk 3 above)
Comment (nikpivkin) [reply pkg/iac/scanners/terraform/scanner_test.go]: Done https://github.com/aquasecurity/trivy/pull/6160/commits/f564f02d7f74ccc4f2d7fdb36aaf18c31fb16b09

File: pkg/iac/scanners/terraform/parser/parser_test.go (hunk 2 above)
Comment (nikpivkin) [reply pkg/iac/scanners/terraform/parser/parser_test.go:1183]: Done https://github.com/aquasecurity/trivy/pull/6160/commits/f564f02d7f74ccc4f2d7fdb36aaf18c31fb16b09

File: pkg/iac/scanners/terraform/parser/modules_test.go (hunk 1 above)
Comment (simar7) [reply pkg/iac/scanners/terraform/parser/modules_test.go:31]: I see, is it possible to have a root module that's remote/registry address?

Comment (nikpivkin) [reply pkg/iac/scanners/terraform/parser/modules_test.go:31]: @simar7 No, the root can only be local

**Review by simar7 — APPROVED** [review]

**Pull request merged**